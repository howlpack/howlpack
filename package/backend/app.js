import Koa from "koa";
import cors from "@koa/cors";
import Router from "@koa/router";
import bodyParser from "koa-bodyparser";
import session from "koa-session";
import HelloRouter from "./api/hello/index.js";
import CryptoRouter from "./api/crypto/index.js";
import BotRouter from "./api/bot/index.js";
import TwitterRouter from "./api/twitter/index.js";
import error from "./middleware/error.js";

const app = new Koa();
app.proxy = Boolean(process.env.API_PROXY);
app.keys = [process.env.KOA_SESSION_SECRET];

const publicRouter = new Router();

app.use(
  cors({
    credentials: "true",
  })
);

app.use(
  session(
    {
      maxAge: 1800000,
      secure: true,
      sameSite: "none",
    },
    app
  )
);

app.use(error());
app.use(bodyParser());

// === public routes ===
publicRouter.redirect("/", "/api/hello");
const publicRoutes = [
  ["/api/hello", HelloRouter.routes(), HelloRouter.allowedMethods()],
  ["/api/crypto", CryptoRouter.routes(), CryptoRouter.allowedMethods()],
  ["/api/bot", BotRouter.routes(), BotRouter.allowedMethods()],
  ["/api/twitter", TwitterRouter.routes(), TwitterRouter.allowedMethods()],
];

publicRoutes.map((r) => publicRouter.use(...r));

app.use(publicRouter.routes());
app.use(publicRouter.allowedMethods());

export default app;
