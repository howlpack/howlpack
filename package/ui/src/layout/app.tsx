import { Outlet } from "react-router-dom";
import {
  Box,
  CssBaseline,
  Grid,
  ThemeProvider,
  useMediaQuery,
} from "@mui/material";
import { Container } from "@mui/system";
import { Helmet } from "react-helmet";
import { Fragment } from "react";
import { darkTheme, lightTheme } from "../lib/theme";
import Header from "./components/header";
import Footer from "./components/footer";
import Menu from "./components/menu";
import Snackbar from "../components/snackbar";

export default function App() {
  const prefersDarkMode = useMediaQuery("(prefers-color-scheme: dark)");

  return (
    <ThemeProvider theme={prefersDarkMode ? darkTheme : lightTheme}>
      <Helmet>
        <title>Howlpack ðŸŽ’</title>
      </Helmet>
      <Snackbar />
      <CssBaseline />

      <Container maxWidth="lg">
        <Header />

        <Fragment>
          <Grid container>
            <Grid item lg={2} sm={3} xs={12}>
              <Box sx={{ position: { sm: "sticky" }, top: { sm: 8 } }}>
                <Menu />
              </Box>
            </Grid>
            <Grid item lg={10} sm={9} xs={12}>
              <Box sx={{ mt: 1, ml: { sm: 5 } }}>
                <Outlet />
              </Box>
            </Grid>
          </Grid>
        </Fragment>
        <Footer />
      </Container>
    </ThemeProvider>
  );
}
