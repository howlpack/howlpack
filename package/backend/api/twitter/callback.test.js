import request from "supertest";

import app from "@howlpack/howlpack-backend/app.js";

test("Twitter callback", async () => {
  const msg = "foo";
  const state = {
    message: msg,
    dens_name: "howlpack",
    address: "juno1sp3wrffwvtsaz255x63w7c5m40glej8p3vwcwt",
    pubKey: "A4i50fakPVd3KGDzGmkKL88j6VcPZlqQ6x32t1sDCMqp",
    signature:
      "pDPhgDKjV82TDvq7IqAKX2LOnCwhuNcCcvMOaS5a1JocMKLZsx4RgH1A9Vz51mLguN+cM6eg2BBZoijJSVdy1Q==",
  };

  const code = "twitter_code";

  const response = await request(app.callback()).get(
    `/api/twitter/callback?state=${btoa(JSON.stringify(state))}&code=${code}`
  );
  expect(response.status).toBe(200);
  expect(response.text).toMatchSnapshot();
});
