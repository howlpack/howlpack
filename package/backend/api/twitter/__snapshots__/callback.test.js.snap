// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello world works 1`] = `"OK"`;

exports[`Twitter callback 1`] = `"OK"`;
